sources/engine/Stride.Physics/Shapes/StaticMeshColliderShape.cs

```diff
@@ -5,65 +5,314 @@
+                        var nodeTransform = nodeTransforms[node.ParentIndex];
+                        Matrix.Multiply(ref localMatrix, ref nodeTransform, out worldMatrix);
+                    }
+                    else
+                    {
+                        worldMatrix = localMatrix;
+                    }
+
+                    if (i != 0)
+                    {
+                        nodeTransforms[i] = worldMatrix;
+                    }
+                }
+            }
+            
+            int totalVerts = 0, totalIndices = 0;
+            foreach (var meshData in model.Meshes)
+            {
+                totalVerts += meshData.Draw.VertexBuffers[0].Count;
+                totalIndices += meshData.Draw.IndexBuffer.Count;
+            }
+            
+            var combinedVerts = new List<Vector3>(totalVerts);
+            var combinedIndices = new List<int>(totalIndices);
+
+            foreach (var meshData in model.Meshes)
+            {
+                var vBuffer = meshData.Draw.VertexBuffers[0].Buffer;
+                var iBuffer = meshData.Draw.IndexBuffer.Buffer;
+                byte[] verticesBytes = null;
+                byte[] indicesBytes = null;
+                
+                // Both of those can fail when data has already been sent to GPU
+                var vertexBufferRef = AttachedReferenceManager.GetAttachedReference(vBuffer);
+                var indexBufferRef = AttachedReferenceManager.GetAttachedReference(iBuffer);
+                if (vertexBufferRef.Data != null)
+                    verticesBytes = ((BufferData)vertexBufferRef.Data).Content;
+                if (indexBufferRef.Data != null)
+                    indicesBytes = ((BufferData)indexBufferRef.Data).Content;
+                
+                // Editor-specific workaround, we can't load assets when the file provider is null,
+                // will most likely break on non-dx11 APIs
+                if((verticesBytes == null || indicesBytes == null) && dbProvider != null && dbProvider.FileProvider == null)
+                {
+                    var flags = System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance;
+                    var commandList = (CommandList)typeof(GraphicsDevice)
+                        .GetField("InternalMainCommandList", flags)
+                        .GetValue(vBuffer.GraphicsDevice);
+                    
+                    verticesBytes = new byte[vBuffer.SizeInBytes];
+                    indicesBytes = new byte[iBuffer.SizeInBytes];
+                    fixed (byte* window = verticesBytes)
+                        FetchBufferFromGPU(vBuffer, commandList, new DataPointer(window, verticesBytes.Length));
+                    fixed (byte* window = indicesBytes)
+                        FetchBufferFromGPU(iBuffer, commandList, new DataPointer(window, indicesBytes.Length));
+                }
+                
```

Comment:
That's a small editor-specific workaround which fetches the model from gpu instead of reading from disk, it would have been an issue with builds not under dx11 but this behavior only occurs in editor.

I had to do it this way because editor games don't have access to the file provider, this is something I would need @xen2 's or someone more familiar with this to comment on; it looks like the provider is mounted and unmounted at specific time, I'm not certain why or how I should notify the editor to mount it for this specific operation.